src/plugins/proxy/src/plugin.cpp

```diff
@@ -14,12 +14,30 @@
 #include "openvino/runtime/device_id_parser.hpp"
 #include "openvino/runtime/iinfer_request.hpp"
 #include "openvino/runtime/iremote_context.hpp"
+#include "openvino/runtime/so_ptr.hpp"
 #include "openvino/util/common_util.hpp"
 #include "plugin.hpp"
 #include "remote_context.hpp"
 
 namespace {
 
+ov::RemoteContext create_proxy_context(const ov::SoPtr<ov::ICompiledModel>& compiled_model,
+                                       const std::string dev_name,
+                                       size_t dev_idx,
+                                       bool has_dev_idx,
+                                       bool is_new_api) {
+    ov::SoPtr<ov::IRemoteContext> device_context;
+    ov::RemoteContext remote_context;
+    try {
+        device_context = {compiled_model->get_context(), compiled_model._so};
```

Comment:
@ilyachur here proxy plugin can construct the context, but auto/batch plugin can't access this constructor.. why icompiled_model base class does not hold IRemoteContext instead of RemoteContext?

Reply:
@songbell Good catch!

Thank you. I updated the API and provide helpers which can be used for meta plugins